src/Npgsql/NpgsqlTracingOptions.cs

```diff
@@ -1,9 +1,50 @@
+using System;
+using System.Diagnostics;
+
 namespace Npgsql;
 
 /// <summary>
 /// Options to configure Npgsql's support for OpenTelemetry tracing.
-/// Currently no options are available.
 /// </summary>
 public class NpgsqlTracingOptions
 {
-}
\ No newline at end of file
+    internal static NpgsqlTracingOptions? Current { get; set; }
+
+    /// <summary>
+    /// Gets or sets a filter function that determines whether or not to
+    /// collect telemetry on a per <see cref="NpgsqlCommand"/> basis.
+    /// </summary>
+    public Func<NpgsqlCommand, bool>? FilterNpgsqlCommand { get; set; }
+
+    /// <summary>
+    /// Gets or sets an action to enrich an <see cref="Activity"/> with <see cref="NpgsqlCommand"/>.
+    /// </summary>
+    public Action<Activity, NpgsqlCommand>? EnrichWithNpgsqlCommand { get; set; }
+
+    /// <summary>
+    /// Gets or sets a function that provides a span's name on a per <see cref="NpgsqlCommand"/> basis.
+    /// </summary>
+    public Func<NpgsqlCommand, string?>? ProvideSpanNameForNpgsqlCommand { get; set; }
+
+    /// <summary>
+    /// Gets or sets a filter function that determines whether or not to
+    /// collect telemetry on a per <see cref="NpgsqlBatch"/> basis.
+    /// </summary>
+    public Func<NpgsqlBatch, bool>? FilterNpgsqlBatch { get; set; }
+
+    /// <summary>
+    /// Gets or sets an action to enrich an <see cref="Activity"/> with <see cref="NpgsqlBatch"/>.
+    /// </summary>
+    public Action<Activity, NpgsqlBatch>? EnrichWithNpgsqlBatch { get; set; }
+
+    /// <summary>
+    /// Gets or sets a function that provides a span's name on a per <see cref="NpgsqlBatch"/> basis.
+    /// </summary>
+    public Func<NpgsqlBatch, string?>? ProvideSpanNameForNpgsqlBatch { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether to disable the "time-to-first-read" event.
+    /// Default is false to preserve existing behavior.
+    /// </summary>
+    public bool DisableFirstResponseEvent { get; set; }
```

Comment:
Note that back when this was first implemented, we had a [big discussion](https://github.com/npgsql/npgsql/pull/3967#discussion_r713658094) on whether we should report time-to-first-read or time-to-last-read; we ended up going with time-to-first-read, but it's by no means clear that this is always the right/interesting thing.

So rather than having a bool for disabling time-to-first-read, maybe we should introduce a 3-value enum instead - disable, first and last. Of course, we don't actually have to implement last right now, but having the enum (with only the first two values) would allow us to implement it later...

Reply:
Yeah, it's just I do not really see a lot of value in having a span to end when we get a first response (just having the event time-to-last-read is pointless since just after it we'll close the span). Technically we're still iterating over the result set and doing work, it's not like users can just drop a data reader without reading every single row. So for now I propose to leave it as a boolean, and in case someone will actually request to have a span to end after we get the response, we'll add another property and deprecate this one.

Reply:
OK, am fine with that.